src/main/kotlin/org/jetbrains/research/transformations/Transformations.kt

```diff
@@ -0,0 +1,36 @@
+package org.jetbrains.research.transformations
+
+import com.github.gumtreediff.tree.ITree
+
+public class Transformations() {
+    public val var_anon : MutableMap<String, String> = mutableMapOf()
+    private var num_var_anon = 0
+
+    // this method should anonymize all variables in code
+    fun anonymize(tree :ITree, toStoreMetadata :Boolean) {
+        if (!tree.isLeaf) {
+            for (node in tree.children) {
+
+                // if type is "Store" (80218305)
+                if (node.type == 80218305){
+                    if (node.label !in var_anon.keys) {
```

Comment:
I think you can make more Kotlin style by using [getOrNull](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/get-or-null.html):

```
anonVars.getOrNull(node.label)?.let {
   your actions
}
```

What do you think?

Reply:
I think the best way is to use **getOrPut** (let's also move out all about setting anon names since it can be thought of as a separated part of code):

```
    private fun setAnonLabel(node: ITree, toStoreMetadata: Boolean) : String {
        val newLabel = "v$num_var_anon"
        if (toStoreMetadata) node.setMetadata(newLabel, node.label)
        node.label = newLabel
        return newLabel
    }
```
so it becomes just `var_anon.getOrPut(node.label, { setAnonLabel(node, toStoreMetadata) })`

Reply:
Of course, you are right. I did not see the assignment action! Thanks!